src/mahoji/lib/abstracted_commands/maniacalMonkeyCommand.ts

```diff
@@ -0,0 +1,315 @@
+		return `Manical monkeys require **175 QP**, and you have ${QP} QP.\n`;
+	}
+
+	const [, , attackStyles] = resolveAttackStyles(user, {
+		monsterID: maniacalMonkeyID
+	});
+
+	const boosts: string[] = [];
+
+	let timeToFinish = 0.24 * Time.Minute;
+	const maxTripLength = calcMaxTripLength(user, 'ManiacalMonkey');
+
+	const FIVE_HOURS = Time.Hour * 5;
+	const kc = Math.max(1, user.getKC(maniacalMonkeyID));
+	// every five hours become 1% better to a cap of 10%
+	const percentReduced = Math.min(Math.floor(kc / (FIVE_HOURS / timeToFinish)), 10);
+	timeToFinish -= (timeToFinish * percentReduced) / 100;
+
+	const [newTime, skillBoostMsg] = applySkillBoost(user, timeToFinish, attackStyles);
+
+	timeToFinish = newTime;
+	boosts.push(skillBoostMsg);
+
+	if (percentReduced >= 1) boosts.push(`${percentReduced}% for KC`);
+
+	// Initialize consumable costs before any are calculated.
+	const consumableCosts: Consumable[] = [];
+
+	// Calculate Chinning and Barrage boosts + costs:
+	if (method === 'barrage' && attackStyles.includes(SkillsEnum.Magic)) {
+		if (user.skillLevel(SkillsEnum.Magic) < 94) {
+			return `You need 94 Magic to use Ice Barrage. You have ${user.skillLevel(SkillsEnum.Magic)}`;
+		}
+		if (user.hasEquippedOrInBank('Occult necklace')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Occult necklace');
+		}
+		if (user.hasEquippedOrInBank('Tormented bracelet')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Tormented bracelet');
+		}
+		if (user.hasEquippedOrInBank('Kodai wand')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Kodai wand');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe top')) {
+			timeToFinish *= (100 - 3) / 100;
+			boosts.push('3% for Ancestral robe top');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe bottom')) {
+			timeToFinish *= (100 - 2) / 100;
+			boosts.push('2% for Ancestral robe bottom');
+		}
+		consumableCosts.push(iceBarrageConsumables);
+		timeToFinish = reduceNumByPercent(timeToFinish, 75);
+		boosts.push('75% for Ice Barrage');
+	} else if (method === 'burst' && attackStyles.includes(SkillsEnum.Magic)) {
+		if (user.skillLevel(SkillsEnum.Magic) < 70) {
+			return `You need 70 Magic to use Ice Burst. You have ${user.skillLevel(SkillsEnum.Magic)}`;
+		}
+		if (user.hasEquippedOrInBank('Occult necklace')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Occult necklace');
+		}
+		if (user.hasEquippedOrInBank('Tormented bracelet')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Tormented bracelet');
+		}
+		if (user.hasEquippedOrInBank('Kodai wand')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Kodai wand');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe top')) {
+			timeToFinish *= (100 - 3) / 100;
+			boosts.push('3% for Ancestral robe top');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe bottom')) {
+			timeToFinish *= (100 - 2) / 100;
+			boosts.push('2% for Ancestral robe bottom');
+		}
```

Comment:
Is there a way that you could prevent repeating the boosts?

Reply:
yea these are quite a bit of boosts. putting them in a data format would be better.